cilium/CFP-36975-configuration-profiles.md

```diff
@@ -0,0 +1,121 @@
+
+## Goals
+
+The main goals of this proposal are:
+*   Define a process for creating a new Configuration Profile
+*   Establish a testing framework to ensure the stability and correctness of each profile.
+*   Enable the community to propose and contribute new profiles.
+
+## Non-Goals
+
+*   Completely eliminate individual feature flags.  Advanced users will still be able to customize their deployments beyond the pre-defined profiles.  Profiles are intended to be a starting point, not a restriction.
+*   Create a profile for every possible combination of flags.  The focus is on common, well-understood use cases.
+*   Guarantee that profiles will cover 100% of user needs. It's recognized there will always be edge cases, but this helps a majority of users.
+
+## Example Profile
+
+*   **High-Scale:**  Optimized for large-scale clusters, with high number (thousands) of nodes and high pod churn rate (hundreds per second). Limited to a set of basic networking and K8s features: pod connectivity, K8s Service and basic observability.
+    *   `--enable-policy=never`
+    *   `--enable-k8s-networkpolicy=false`
+    *   `--enable-cilium-network-policy=false`
+    *   `--enable-cilium-clusterwide-network-policy=false`
+    *   `--identity-allocation-mode=crd`
+    *   `--disable-endpoint-crd=true`
+    *   `--enable-cilium-endpoint-slice=false`
+
+    Note: This is only an example profile. It's not ready for use.
+
+## Creating a Profile
+
+*   Provide clear documentation for each profile, including:
+    *   Purpose and use case
+    *   Benefits and limitations (trade-offs)
+    *   Specific configuration options (flags) enabled/disabled
+    *   Installation instructions
+    *   Supported Cilium versions
+    *   Known issues
+*   Implement profile selection via Helm charts.
+
+## Implementation
+
+The implementation will leverage Cilium's Helm chart capabilities:
+
+1.  **Helm Chart Values:**  Each profile will be defined as a set of Helm values overrides.  These overrides will configure the necessary Cilium feature flags.  For example, the `values.yaml` for the "High-Scale" profile might include:
+
+    ```yaml
+    # profiles/high-scale.yaml
+    disableEndpointCRD: false
+    policyEnforcementMode: "never"
+    # ... other relevant settings ...
+    ```
+
+2.  **Profile Selection:** Users will select a profile during installation by referencing the appropriate values file:
+
+    ```bash
+    helm install cilium cilium/cilium --version 1.17.x \
+      --namespace kube-system \
+      --values profiles/high-scale.yaml
+      # Or:
+      # --set profile=high-scale  (if we add a 'profile' value to the main chart)
+    ```
+    The main `values.yaml` might include a top-level `profile` key, allowing users to select a profile with `--set profile=high-scale`.  This would then conditionally include the settings from the corresponding profile file.
```

Comment:
I'd prefer the first option (`--values profile/high-scale.yaml`) since it makes it easier for people the option to override specific settings if they know what they're doing (just add more overrides after using `--values` or `--set`)

Reply:
I agree. It makes more sense. I will keep the `--set. profile=` in the design as an alternative for now, until we get broader feedback.

Reply:
I removed the alternative part that suggests also potentially using `--set profile=x`